repository.py

```diff
@@ -821,12 +821,59 @@ def installFromYum(targets, mounts, progress_callback, cachedir):
         rv = p.wait()
         stderr.seek(0)
         stderr = stderr.read()
+        gpg_uncaught_error = 0
```

Comment:
I cannot recall precisely, but:
- (likely original reason) to keep the structure of the existing code; notably, there was already an exception raised with a generic message, and all that needed to be changed was the message
- in line with that, it allows to keep the "yum exited" log, which would have to be duplicated, or wrapped with the exception in a helper (and delegating the raise to a helper function causes its own problems)
- additionally it allows to make sure to use the most pertinent message, regardless of the matching order, which in turn is constrained by the order in which strings appear.  Maybe not really necessary for this set of messages, but considering this is ad-hoc parsing of a program's output, including parsing of its crashes, it may be useful to retain this flexibility

Reply:
Why instead of using a lot of flags variable you don't just set a string variable `errmsg` to save the specific error and use it? You could initialize it at default generic message and change accordingly based on the specific error.